opentelemetry-sdk/src/trace/provider.rs

```diff
@@ -62,7 +62,7 @@ impl TracerProvider {
     }
 
     /// Span processors associated with this provider
-    pub fn span_processors(&self) -> &Vec<Box<dyn SpanProcessor>> {
+    pub fn span_processors(&self) -> &[Box<dyn SpanProcessor>] {
```

Comment:
```suggestion
    pub(crate) fn span_processors(&self) -> &[Box<dyn SpanProcessor>] {
```

Don't think this method should be public? Unrelated to this PR, but sharing here, as this is considered breaking given this method is public. We should audit all public API and trim everything unnecessary to avoid issues as we move towards stable.

Reply:
+1 let's make it `pub(crate)`

Reply:
lets do the full audit and cleanup as a follow up to keep scope in control for this PR.